src/domain/map/components/HankeSidebar/HankeSidebar.tsx

```diff
@@ -1,42 +1,106 @@
-import React from 'react';
+import React, { ReactNode } from 'react';
 import { useTranslation } from 'react-i18next';
-import { Drawer, DrawerBody, DrawerContent, DrawerOverlay } from '@chakra-ui/react';
+import { Box, Drawer, DrawerBody, DrawerContent, DrawerOverlay } from '@chakra-ui/react';
 import { IconCross } from 'hds-react/icons';
 import Text from '../../../../common/components/text/Text';
-import { formatToFinnishDate } from '../../../../common/utils/date';
-import { HankeData } from '../../../types/hanke';
+import { HankeData, HankeYhteystieto } from '../../../types/hanke';
 import styles from './HankeSidebar.module.scss';
-import HankeIndexes from '../../../hanke/hankeIndexes/HankeIndexes';
+import { formatToFinnishDate } from '../../../../common/utils/date';
+import { TFunction } from 'i18next';
+import HaittaIndexes from '../../../common/haittaIndexes/HaittaIndexes';
+import api from '../../../api/api';
+import { useQuery } from 'react-query';
+import axios from 'axios';
+import useHankeViewPath from '../../../hanke/hooks/useHankeViewPath';
+import { ROUTES } from '../../../../common/types/route';
+import { Link } from 'hds-react';
+
+function omistajaNimet(
+  omistajat: HankeYhteystieto[],
+  t: TFunction<'translation', undefined>,
+): string {
+  const names = omistajat.map((yhteystieto) =>
+    yhteystieto.nimi == undefined || yhteystieto.nimi.length > 0
+      ? yhteystieto.nimi
+      : t('hankeSidebar:labels:yksityishenkilo'),
+  );
+  const uniqueNames = Array.from(new Set(names));
+  return uniqueNames.join(', ');
+}
+
+const getHankkeet = async () => {
```

Comment:
This is the same function as in `src/domain/hanke/portfolio/HankePortfolioContainer.tsx` and could be shared. Also `useHankkeet` hook could probably be shared with `HankePortfolioContainer`.

Reply:
I created `useHankkeet` hook under `src/domain/hanke/hooks` and use it in both places.